packages/flutter/test/material/dialog_theme_test.dart

```diff
@@ -128,4 +133,38 @@ void main() {
       skip: !Platform.isLinux,
     );
   });
+
+  testWidgets('Custom Title Text Style', (WidgetTester tester) async {
```

Comment:
This test and the one below should test the 3 sources for the text style: Theme, DialogTheme, constructor parameter.

Reply:
I'm not sure if this is what you were asking for, but I wasn't sure how to incorporate all 3 of those things in each test. I split the `'Custom Title Text Style'` and `'Custom Content Text Style'` tests into 3 different tests each. That way I'm testing 
1. Constructor Param
1. Dialog Theme
1. Theme.textTheme

For both `titleTextStyle` and  `contentTextStyle` (a total of 6 tests).

Reply:
I would have written two tests like this:
```
testWidgets('Custom Title Text Style', (WidgetTester tester) async {
  Widget buildFrame({ TextStyle themeStyle, TextStyle dialogThemeStyle, TextStyle dialogStyle }) {
    return _appWithAlertDialog(
      tester,
      AlertDialog(
        title: Text(titleText),
        titleTextStyle: titleTextStyle,
        actions: <Widget>[ ],
      ),
      theme: ThemeData(
        textTheme: const TextTheme(title: themeStyle)
        dialogTheme: const DialogTheme(titleTextStyle: dialogStyle)
      ),
    );
  }

  await tester.pumpWidget(
    buildFrame(
      themeStyle: const TextStyle(color: Colors.pink),
      dialogThemeStyle: const TextStyle(color: Colors.green),
      dialogStyle: const TextStyle(color: Colors.blue),
    ),
  );
  await tester.tap(find.text('X'));
  await tester.pumpAndSettle();
  final RenderParagraph title = _getTextRenderObjectFromDialog(tester, titleText);
  expect(title.text.style, const TextStyle(color: Colors.blue));

  await tester.pumpWidget(
    buildFrame(
      themeStyle: const TextStyle(color: Colors.pink),
      dialogThemeStyle: const TextStyle(color: Colors.green),
    ),
  );
  await tester.tap(find.text('X'));
  await tester.pumpAndSettle();
  final RenderParagraph title = _getTextRenderObjectFromDialog(tester, titleText);
  expect(title.text.style, const TextStyle(color: Colors.green));

  await tester.pumpWidget(
    buildFrame(
      themeStyle: const TextStyle(color: Colors.pink),
    ),
  );
  await tester.tap(find.text('X'));
  await tester.pumpAndSettle();
  final RenderParagraph title = _getTextRenderObjectFromDialog(tester, titleText);
  expect(title.text.style, const TextStyle(color: Colors.pink));

});
```

Reply:
ohh I see now, in general I personally prefer to opt for smaller tests: go/tott-520